lib/io/dart_io.cc

```diff
@@ -4,21 +4,31 @@
 
 #include "flutter/lib/io/dart_io.h"
 
+#include "flutter/fml/logging.h"
+
 #include "third_party/dart/runtime/include/bin/dart_io_api.h"
 #include "third_party/dart/runtime/include/dart_api.h"
 #include "third_party/tonic/converter/dart_converter.h"
+#include "third_party/tonic/logging/dart_error.h"
 
 using tonic::ToDart;
+using tonic::LogIfError;
 
 namespace flutter {
 
-void DartIO::InitForIsolate() {
+void DartIO::InitForIsolate(bool disable_http) {
   Dart_Handle result = Dart_SetNativeResolver(
       Dart_LookupLibrary(ToDart("dart:io")), dart::bin::LookupIONative,
       dart::bin::LookupIONativeSymbol);
-  if (Dart_IsError(result)) {
-    Dart_PropagateError(result);
-  }
+  FML_CHECK(!LogIfError(result));
+
+  // The SDK expects this field to represent "allow http" so we switch the
+  // value.
+  Dart_Handle allow_http_value = disable_http ? Dart_False() : Dart_True();
```

Comment:
If the SDK uses "allow", why use "disable" here?  The inconsistency (especially in C++ where there are not named parameters) increases the likelihood for a potential mismatch.

(The style guide for the flutter/flutter repo [discourages negative names](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-double-negatives-in-apis).  Granted, that's not a style guide for flutter/engine, but I think it's still a good convention to follow.)

Reply:
I struggled with this too. The problem is that boolean switches in engine currently don't test for values. They simply infer `true` if the switch exists and `false` if it does not. This works well only for settings whose default value is `false`. 

I would ideally have a switch for `allow_http` and then be able to do `--noallow_http` from the embedder. My understanding is that that would require a large-ish change to the switch processing (and it is only relevant for Android). I figured it is not worth doing especially with many examples of negative names in engine already. 

Another simple way to solve this is to have an explicit switch for `--noallow_http`. But that doesn't really solve the problem. You still end up with a negative name and the negation is pushed to somewhere else.